src/Sarif/ExtensionMethods.cs

```diff
@@ -37,6 +37,11 @@ public static IEnumerable<SarifLog> Split(this SarifLog sarifLog, SplittingStrat
                     partitionFunction = (result) => result.Run.GetHashCode().ToString();
                     break;
                 }
+                case SplittingStrategy.PerRunPerTarget:
+                {
+                    partitionFunction = (result) => result.Locations.FirstOrDefault().PhysicalLocation.ArtifactLocation.Uri.OriginalString;
```

Comment:
>result.Locations [](http://example.com/codeflow?start=52&length=16)

a result without location = Locations = null or Locations = []?

either way, please, add one test for that